lib/rspec/mocks/method_double.rb

```diff
@@ -261,6 +261,15 @@ def definition_target
 
     private
 
+      def method_owner
+        @method_owner ||=
+          if object.kind_of?(Object)
+            Object.instance_method(:method).bind(object).call(@method_name).owner
```

Comment:
This condition fixes for Ruby 1.9.2, 1.9.3, 2.0.0, 2.1.0, and 2.2.10. This is probably because `SimpleDelegator` is a `BasicObject` and cannot bind to `Object` in those cases. Not sure why it did work for Ruby 2.3+ though, even when that's a good thing.

Reply:
Do you mind adding a code comment to clarify that?
We'll [drop support for old Rubies this year](https://github.com/rspec/rspec/issues/61#issuecomment-687290045), so maybe this workaround won't be needed anymore.

Reply:
Yes, I actually added a comment after I left this comment, and after seeing your comment I tried to make the comment even more clear, so thanks for asking.

In https://github.com/rspec/rspec-mocks/pull/1218#issuecomment-392540214 @JonRowe also said that Ruby 2.2 is unlikely to be dropped so we probably still need this patch for a long time.

On the other hand, I really can't get Ruby 1.8.7 work so at least that should be dropped. Now I really want to get this merged so I also started looking into how to get JRuby pass, and I am running this version of JRuby and everything actually just passed:

```
jruby 9.2.13.0 (2.5.7) 2020-08-03 9a89c94bcc OpenJDK 64-Bit Server VM 14.0.2+12 on 14.0.2+12 +jit [linux-x86_64]
```

If we're ok to upgrade to newer JRuby, I hope we can just do that because it's likely older JRuby bugs and I really don't want to add even more workaround :(